### PR TITLE
Aman/sort filter

### PR DESCRIPTION
**Pull request by ataddesse**
# Description

- Modifies the leads component code to use ngAfterViewInit
- Got the filter field working, but still needs modification in the next PR.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: frontend/src/app/crm/leads/leads.component.html [hunk 1]
```diff
@@ -1,43 +1,79 @@
...
+
+    <mat-table [dataSource]="dataSource"
+               matSort
+               cdkFocusInitial
+               class="mat-elevation-z8">
+
+      <!-- Lead Id Column -->
+      <ng-container matColumnDef="lead_id">
+        <mat-header-cell *matHeaderCellDef
+                         mat-sort-header> ID </mat-header-cell>
```
Comment (r0ddy) [inline frontend/src/app/crm/leads/leads.component.html:27]: Could you change this to Lead Id?

File: frontend/src/app/crm/leads/leads.component.html [hunk 2]
```diff
@@ -1,43 +1,79 @@
...
+        <mat-cell *matCellDef="let lead"> {{lead.lead_id}} </mat-cell>
+      </ng-container>
+
+      <!-- Name Column -->
+      <ng-container matColumnDef="name">
+        <mat-header-cell *matHeaderCellDef
+                         mat-sort-header> Name </mat-header-cell>
```
Comment (r0ddy) [inline frontend/src/app/crm/leads/leads.component.html:34]: Could you change this to Lead Name?

File: frontend/src/app/crm/leads/leads.component.ts [hunk 3]
```diff
@@ -1,43 +1,64 @@
...
+  loadAllLeads(): void {
+    this.isLoading$.next(true);
+    this.leadService.getAllLeads().pipe(first()).subscribe((leads) => {
+      this.dataSource.data = leads;
+      this.isLoading$.next(false);
+    });
+  }
+  applyFilter(event: Event) {
+    // console.log(this.dataSource);
+    console.log('Hello1')
```
Comment (r0ddy) [inline frontend/src/app/crm/leads/leads.component.ts:52]: Delete all these console.log calls.

File: frontend/src/app/crm/leads/leads.component.html (hunk 2 above)
Comment (ataddesse) [reply frontend/src/app/crm/leads/leads.component.html:34]: Ight

File: frontend/src/app/crm/leads/leads.component.html (hunk 1 above)
Comment (ataddesse) [reply frontend/src/app/crm/leads/leads.component.html:27]: Ight

File: frontend/src/app/crm/leads/leads.component.ts (hunk 3 above)
Comment (ataddesse) [reply frontend/src/app/crm/leads/leads.component.ts:52]: Ight

**Pull request merged**